File: Python/02. Working with Numbers/43. Number is expressed as a sum of two prime numbers.py
#  Goldbech's Conjecture Logic
number = int(input("Enter a number: "))
res = False
lst = []
for i in range(2, number):
    if all( i % j != 0 for j in range(2,i)):
        lst.append(i)
for k in range( len(lst) ):
    for l in range( k,len(lst) ):
        if ( lst[k] + lst[l] == number ):
            print( lst[k] ,"+", lst[l],"=",number)
            res = True
if not res:
    print(number, "cannot be expressed as a sum of two prime numbers")

File: Java/02. Working with Numbers/43. Number is expressed as a sum of two prime numbers.java
// Goldbech's Conjecture Logic
import java.util.Scanner;

class Main {
    public static void main(String[] args) {
        Scanner userInput = new Scanner(System.in);

        System.out.println("Enter a Number:");
        int number = userInput.nextInt();
        boolean res = false;

        for (int i=2; i <= number / 2; i++) {
            boolean prime = true;
            for (int j=2; j < i;j++) {
                if(i%j == 0) {
                    prime = false;
                }
            }
            int secondnum = number - i;
            boolean isPrime = true;
            for( int j=2; j<secondnum;j++) {
                if(secondnum % j == 0) {
                    isPrime = false;
                }
            }
            
            if(prime && isPrime) {
                System.out.println(number + " = " + i + " + " + (secondnum));
                res = true;
            }
        }
        if ( !res ) {
            System.out.println(number + " cannot be expressed as a sum of two prime numbers");
        }
        userInput.close();
    }
}